community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -1297,4 +1301,9 @@ private void closeCursor()
             }
         }
     }
+
+    public boolean isDirty()
```

Comment:
Misleading name since state could have changed since startup, `wasDirtyOnStartup` or something to that effect. Also, this functionality already exist through the `Monitor`. If we want to rely on that we can edit `private GBPTree.Monitor treeMonitor()` in `NativeLabelScanStore` and `NativeSchemaNumberIndex`.

Reply:
no i will not gonna use monitor for that, its not a monitor business to return you some kind of object state, that just wrong

Reply:
Alright, just the name change then 😄